Sgrep -> Semgrep in ocaml part of semgrep (semgrep-core/)

aryx opened the pull request.

aryx committed:
* semgrep-core/: sgrep -> semgrep for files and dune

aryx committed:
sgrep -> semgrep inn ocaml files for modules

This closes ossie #105

Test plan:
make test

aryx committed:
* semgrep-core/tests/dune: fix

brendongo commented:
Build is failing on `File "tests/test.ml", line 82, characters 4-23: Error: Unbound module Sgrep_generic`

aryx committed:
* semgrep-core/tests/test.ml: missing Sgrep_generic ref.
Did not catch them at compilation time because of previous make install
that put those old sgrep_xxx lib in OPAM.
ARGH

test plan:
rm -rf ~/.opam/lib/4.07.1/sgrep_*
make

aryx committed:
update path to main_sgrep.exe

aryx commented:
should close https://github.com/returntocorp/enterprise/issues/105
